bindings_node/src/mls_client.rs

```diff
@@ -182,14 +182,14 @@ impl NapiClient {
 
     // check for missing signatures
     let missing_signatures = signature_request.missing_signatures();
-    let missing = missing_signatures
+    let signatures_to_add = missing_signatures
       .iter()
-      .filter(|id| !self.signatures.contains_key(id))
+      .filter(|id| self.signatures.contains_key(id))
```

Comment:
You will need to add all missing_signatures, not only the ones in the `self.signatures`

Reply:
this function will error out if a signature is missing, which is intentional.

am i missing something?

Reply:
`filter` won't return error, it's just filtering out missing signatures that are not in the `self.signatures` hash map

Reply:
yep, the error will come from `self.inner_client.register_identity(signature_request)`

Reply:
https://doc.rust-lang.org/src/core/iter/traits/iterator.rs.html#883-886

Reply:
ok then I guess you don't need to filter, just add_signature() from self.signatures and it will error if the signature is not needed

Reply:
oh, that's even better! i'll update the logic again.

Reply:
does this update look good? [Simplify signature logic](https://github.com/xmtp/libxmtp/pull/808/commits/746fcc8fa125d41a6dcdd09f780a32c843a50df0)

Reply:
yep